include/wlr/types/wlr_cursor.h

```diff
@@ -0,0 +1,113 @@
+#ifndef _WLR_TYPES_CURSOR_H
+#define _WLR_TYPES_CURSOR_H
+#include <wayland-server.h>
+#include <wlr/types/wlr_output.h>
+#include <wlr/types/wlr_output_layout.h>
+#include <wlr/types/wlr_input_device.h>
+#include <wlr/types/wlr_geometry.h>
+#include <wlr/xcursor.h>
+
+struct wlr_cursor_state;
+//struct wlr_cursor_impl *;
+
+struct wlr_cursor {
+    struct wlr_cursor_state *state;
+    //struct wlr_cursor_impl *impl;
+    int x, y;
+
+    struct {
+        struct wl_signal motion;
+		struct wl_signal motion_absolute;
+		struct wl_signal button;
+		struct wl_signal axis;
+
+		struct wl_signal touch_up;
+		struct wl_signal touch_down;
+		struct wl_signal touch_motion;
+		struct wl_signal touch_cancel;
+
+		struct wl_signal tablet_tool_axis;
+		struct wl_signal tablet_tool_proximity;
+		struct wl_signal tablet_tool_tip;
+		struct wl_signal tablet_tool_button;
+    } events;
```

Comment:
This indentation is fucky

Reply:
Also, I think it would make sense for wlr_cursor to expose a pointer-like API for all of the devices its using. IMO exposing touch/tool events like this is a bit of a non-sequiter since using touch or tool devices normally will completely circumvent the wayland pointer API this is meant to abstract.

Reply:
so are you suggesting to merge some of the events together?

Reply:
No, on further thought this approach is fine.